core/src/main/java/org/apache/calcite/sql/validate/SqlValidator.java

```diff
@@ -912,15 +912,39 @@ interface Config {
      */
     Config withLenientOperatorLookup(boolean lenient);
 
-    /** Returns whether the validator allows measures to be used without the
-     * AGGREGATE function. Default is true. */
-    @Value.Default default boolean nakedMeasures() {
+    /** Returns whether the validator allows measures to be used without
+     * AGGREGATE function outside an aggregating query. Default is true.
+     */
+    @Value.Default default boolean nakedMeasuresOutsideAggregatingQuery() {
       return true;
     }
 
+    /** Sets whether the validator allows measures to be used without AGGREGATE
+     * function outside an aggregating query.
+     */
+    Config withNakedMeasuresOutsideAggregatingQuery(boolean value);
+
+    /** Returns whether the validator allows measures to be used without
+     * AGGREGATE function inside an aggregating query. Default is true.
+     */
+    @Value.Default default boolean nakedMeasuresInsideAggregatingQuery() {
+      return true;
+    }
+
+    /** Sets whether the validator allows measures to be used without AGGREGATE
+     * function inside an aggregating query.
+     */
+    Config withNakedMeasuresInsideAggregatingQuery(boolean value);
+
     /** Sets whether the validator allows measures to be used without the
-     * AGGREGATE function. */
-    Config withNakedMeasures(boolean nakedMeasures);
+     * AGGREGATE function inside or outside aggregate queries.
+     * Deprecated: use the inside / outside variants instead.
+     */
+    @Deprecated
+    default Config withNakedMeasures(boolean nakedMeasures) {
```

Comment:
Set a release for removal. I suggest `// to be removed before 1.38`. Measures are an experimental feature, so one minor release notice is sufficient.